varname.py

```diff
@@ -19,18 +19,36 @@
 class VarnameRetrievingError(Exception):
     """When failed to retrieve the varname"""
 
-def varname(caller: int = 1, raise_exc: bool = True) -> Optional[str]:
+def varname(caller: int = 1,
+            nvars: int = 1,
+            raise_exc: bool = True) -> Optional[Union[str, Tuple[str]]]:
     """Get the variable name that assigned by function/class calls
 
     Args:
         caller: The call stack index, indicating where this function
             is called relative to where the variable is finally retrieved
+        nvars: Number of variables to be expected on left-hand side.
+            if nvars > 1, a tuple of variable names will be returned.
+            Note that you also have to return the same number of elements in
+            you function:
+            For example:
+            >>> from varname import varname
+            >>> def func():
+            >>>   print(varname(nvars=2))
+            >>>   return 1, 2
+            >>> a, b = func() # prints ('a', 'b')
+            >>> # a == 1 and b == 2
+
+            if also can be `None`, then you have to handle the return values
+            by yourself.
         raise_exc: Whether we should raise an exception if failed
             to retrieve the name.
 
     Returns:
         The variable name, or `None` when `raise_exc` is `False` and
             we failed to retrieve the variable name.
+        A tuple of variable names on LHS if nvars > 1 or nvars is None and
+        multiple variables on RHS.
```

Comment:
1. The docs should be crystal clear that this means `if (nvars > 1) or (nvars is None and multiple variables on RHS)`
2. It's actually not quite true, it's really `if (nvars > 1) or (nvars is None and RHS is a list/tuple)`. Code like `(foo,) = stuff()` is real, see https://blog.ram.rachum.com/post/1198230058/python-idiom-for-taking-the-single-item-from-a
3. I think it should be `if (nvars > 1) or (nvars is None)`, period. It's confusing that that it might return a tuple or string and you have to check. People are usually going to iterate over the result, and if they test it with a single variable with a one-letter name it'll seem to work, then their code will break when a longer variable name is used.

Reply:
Jesus, you caught me.
I meant that but wrote it in a wrong way.

Reply:
And I actually had a typo for that `RHS`, which I actually meant `LHS`. But you are right,
> if (nvars > 1) or (nvars is None), period.

is the right statement.

Reply:
It's an interesting idea but I'm not convinced. I can't see why a user of varname would care whether the single variable being assigned to is in a tuple or not. That's like asking "is my caller asserting that a sequence of one value is returned?" If they're such power users that they want that level of detail then they can use executing and access the AST. varname should be a more user friendly wrapper.

Reply:
Well then, do you think "multiple variables" is the right term? Because this is the purpose for this PR:
`a, b = func()` to get `varname` inside to return `('a', 'b')`

Reply:
Perhaps I was unclear, I think the behaviour (not just the docs) should be changed to `if (nvars > 1) or (nvars is None)`.

Reply:
I assume you mean `varname` returns a tuple `if (nvars > 1) or (nvars is None)`. Then does this include this case:
```python
def func():
  return varname(nvars=None)
a = func()
```
Then do you mean `a` should be `('a', )` instead of `'a'`?

Reply:
Yes

Reply:
Hmmm, looks like it makes sense to me now. Since by passing `nvars` they should know what is being expected from LHS.
If you want a single variable, they should use `1`

Reply:
Agreed not mixing `1` and `None` for the argument.